code/modules/food_and_drinks/food/pizza.dm

```diff
@@ -0,0 +1,361 @@
+
+/obj/item/food/pizzaslice/vegetable
+	name = "vegetable pizza slice"
+	desc = "A slice of the most green pizza of all pizzas not containing green ingredients."
+	icon_state = "vegetablepizzaslice"
+	tastes = list("crust" = 1, "tomato" = 2, "cheese" = 1, "carrot" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+
+/obj/item/food/pizza/vegetable/raw
+	name = "raw vegetable pizza"
+	icon_state = "vegetablepizza_raw"
+	foodtypes =  GRAIN | VEGETABLES | DAIRY | RAW
+	burns_in_oven = FALSE
+	slice_type = null
+
+/obj/item/food/pizza/vegetable/raw/MakeBakeable()
+	AddComponent(/datum/component/bakeable, /obj/item/food/pizza/vegetable, rand(70 SECONDS, 80 SECONDS), TRUE, TRUE)
+
+/obj/item/food/pizza/donkpocket
+	name = "donkpocket pizza"
+	desc = "Who thought this would be a good idea?"
+	icon_state = "donkpocketpizza"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 20, /datum/reagent/consumable/tomatojuice = 6, /datum/reagent/medicine/omnizine = 10, /datum/reagent/consumable/nutriment/vitamin = 5)
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "meat" = 1, "laziness" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY | MEAT | JUNKFOOD
+	slice_type = /obj/item/food/pizzaslice/donkpocket
+
+/obj/item/food/pizza/donkpocket/Initialize()
+	. = ..()
+	AddElement(/datum/element/processable, TOOL_KNIFE, /obj/item/food/pizzaslice/donkpocket, 6, 30)
+
+/obj/item/food/pizza/donkpocket/raw
+	name = "raw donkpocket pizza"
+	icon_state = "donkpocketpizza_raw"
+	foodtypes =  GRAIN | VEGETABLES | DAIRY | MEAT | JUNKFOOD | RAW
+	burns_in_oven = FALSE
+	slice_type = null
+
+/obj/item/food/pizza/donkpocket/raw/MakeBakeable()
+	AddComponent(/datum/component/bakeable, /obj/item/food/pizza/donkpocket, rand(70 SECONDS, 80 SECONDS), TRUE, TRUE)
+
+/obj/item/food/pizzaslice/donkpocket
+	name = "donkpocket pizza slice"
+	desc = "Smells like donkpocket."
+	icon_state = "donkpocketpizzaslice"
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "meat" = 1, "laziness" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY | MEAT | JUNKFOOD
+
+/obj/item/food/pizza/dank
+	name = "dank pizza"
+	desc = "The hippie's pizza of choice."
+	icon_state = "dankpizza"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 25, /datum/reagent/consumable/doctor_delight = 5, /datum/reagent/consumable/tomatojuice = 6, /datum/reagent/consumable/nutriment/vitamin = 5)
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "meat" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+	slice_type = /obj/item/food/pizzaslice/dank
+
+/obj/item/food/pizza/dank/raw
+	name = "raw dank pizza"
+	icon_state = "dankpizza_raw"
+	foodtypes =  GRAIN | VEGETABLES | DAIRY | RAW
+	burns_in_oven = FALSE
+	slice_type = null
+
+/obj/item/food/pizza/dank/raw/MakeBakeable()
+	AddComponent(/datum/component/bakeable, /obj/item/food/pizza/dank, rand(70 SECONDS, 80 SECONDS), TRUE, TRUE)
+
+/obj/item/food/pizza/dank/Initialize()
```

Comment:
Redundant due to the base typepath having a generic method, would actually create double.